{"uid":"961b50eabb5165e1","name":"test_edit_user_invalid_username","fullName":"tests.api.test_api_edit_user#test_edit_user_invalid_username","historyId":"8ad617eccc3621185820649064ab2622","time":{"start":1718059349363,"stop":1718059349397,"duration":34},"status":"failed","statusMessage":"AssertionError: Expected error message for invalid username\nassert \"The user doesn't exist\" in '404 Client Error:  for url: http://localhost:4001/users/nonexistent_user'\n +  where '404 Client Error:  for url: http://localhost:4001/users/nonexistent_user' = str(HTTPError('404 Client Error:  for url: http://localhost:4001/users/nonexistent_user'))","statusTrace":"edit_user_api = <api.edit_user.EditUser object at 0x7fea9d304140>\nuser_token = 'eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ3ZHVkbGV5IiwiYXV0aCI6W3siYXV0aG9yaXR5IjoiUk9MRV9BRE1JTiJ9LHsiYXV0aG9yaXR5IjoiUk9MRV9DTElFTlQifV0sImlhdCI6MTcxODA1OTM0OSwiZXhwIjoxNzE4MDU5NjQ5fQ.QNAEY2kjsnqGwRvENJnjOMN_cD6lI3VKcbVsvtkVob0'\nnew_user_data = EditUserDto(email='newemail@example.com', firstName='NewFirstName', lastName='NewLastName', roles=['ROLE_ADMIN'])\n\n    def test_edit_user_invalid_username(edit_user_api, user_token, new_user_data):\n        try:\n            invalid_username = \"nonexistent_user\"\n>           response = edit_user_api.api_call(invalid_username, new_user_data, user_token)\n\ntests/api/test_api_edit_user.py:89: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \napi/edit_user.py:8: in api_call\n    response = self.make_request(\"PUT\", f\"users/{username}\", json=user_dict, headers=headers)\napi/base_api.py:33: in make_request\n    response.raise_for_status()\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <Response [404]>\n\n    def raise_for_status(self):\n        \"\"\"Raises :class:`HTTPError`, if one occurred.\"\"\"\n    \n        http_error_msg = \"\"\n        if isinstance(self.reason, bytes):\n            # We attempt to decode utf-8 first because some servers\n            # choose to localize their reason strings. If the string\n            # isn't utf-8, we fall back to iso-8859-1 for all other\n            # encodings. (See PR #3538)\n            try:\n                reason = self.reason.decode(\"utf-8\")\n            except UnicodeDecodeError:\n                reason = self.reason.decode(\"iso-8859-1\")\n        else:\n            reason = self.reason\n    \n        if 400 <= self.status_code < 500:\n            http_error_msg = (\n                f\"{self.status_code} Client Error: {reason} for url: {self.url}\"\n            )\n    \n        elif 500 <= self.status_code < 600:\n            http_error_msg = (\n                f\"{self.status_code} Server Error: {reason} for url: {self.url}\"\n            )\n    \n        if http_error_msg:\n>           raise HTTPError(http_error_msg, response=self)\nE           requests.exceptions.HTTPError: 404 Client Error:  for url: http://localhost:4001/users/nonexistent_user\n\n/opt/hostedtoolcache/Python/3.12.1/x64/lib/python3.12/site-packages/requests/models.py:1021: HTTPError\n\nDuring handling of the above exception, another exception occurred:\n\nedit_user_api = <api.edit_user.EditUser object at 0x7fea9d304140>\nuser_token = 'eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ3ZHVkbGV5IiwiYXV0aCI6W3siYXV0aG9yaXR5IjoiUk9MRV9BRE1JTiJ9LHsiYXV0aG9yaXR5IjoiUk9MRV9DTElFTlQifV0sImlhdCI6MTcxODA1OTM0OSwiZXhwIjoxNzE4MDU5NjQ5fQ.QNAEY2kjsnqGwRvENJnjOMN_cD6lI3VKcbVsvtkVob0'\nnew_user_data = EditUserDto(email='newemail@example.com', firstName='NewFirstName', lastName='NewLastName', roles=['ROLE_ADMIN'])\n\n    def test_edit_user_invalid_username(edit_user_api, user_token, new_user_data):\n        try:\n            invalid_username = \"nonexistent_user\"\n            response = edit_user_api.api_call(invalid_username, new_user_data, user_token)\n            # This line should ideally never be reached if the API behaves correctly\n            assert response.status_code == 404, \"Expected status code 404 for nonexistent user\"\n        except Exception as exc:\n            assert \"404\" in str(exc), \"Expected a 404 status code for invalid username\"\n>           assert \"The user doesn't exist\" in str(exc), \"Expected error message for invalid username\"\nE           AssertionError: Expected error message for invalid username\nE           assert \"The user doesn't exist\" in '404 Client Error:  for url: http://localhost:4001/users/nonexistent_user'\nE            +  where '404 Client Error:  for url: http://localhost:4001/users/nonexistent_user' = str(HTTPError('404 Client Error:  for url: http://localhost:4001/users/nonexistent_user'))\n\ntests/api/test_api_edit_user.py:94: AssertionError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"_session_faker","time":{"start":1718059348061,"stop":1718059348079,"duration":18},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"sign_in_api","time":{"start":1718059348079,"stop":1718059348079,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"user_credentials","time":{"start":1718059348079,"stop":1718059348080,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"edit_user_api","time":{"start":1718059348079,"stop":1718059348079,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"sign_up_api","time":{"start":1718059348079,"stop":1718059348079,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"delete_user_api","time":{"start":1718059348080,"stop":1718059348080,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"user_token","time":{"start":1718059348080,"stop":1718059349275,"duration":1195},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"new_user_data","time":{"start":1718059349275,"stop":1718059349275,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false}],"testStage":{"status":"failed","statusMessage":"AssertionError: Expected error message for invalid username\nassert \"The user doesn't exist\" in '404 Client Error:  for url: http://localhost:4001/users/nonexistent_user'\n +  where '404 Client Error:  for url: http://localhost:4001/users/nonexistent_user' = str(HTTPError('404 Client Error:  for url: http://localhost:4001/users/nonexistent_user'))","statusTrace":"edit_user_api = <api.edit_user.EditUser object at 0x7fea9d304140>\nuser_token = 'eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ3ZHVkbGV5IiwiYXV0aCI6W3siYXV0aG9yaXR5IjoiUk9MRV9BRE1JTiJ9LHsiYXV0aG9yaXR5IjoiUk9MRV9DTElFTlQifV0sImlhdCI6MTcxODA1OTM0OSwiZXhwIjoxNzE4MDU5NjQ5fQ.QNAEY2kjsnqGwRvENJnjOMN_cD6lI3VKcbVsvtkVob0'\nnew_user_data = EditUserDto(email='newemail@example.com', firstName='NewFirstName', lastName='NewLastName', roles=['ROLE_ADMIN'])\n\n    def test_edit_user_invalid_username(edit_user_api, user_token, new_user_data):\n        try:\n            invalid_username = \"nonexistent_user\"\n>           response = edit_user_api.api_call(invalid_username, new_user_data, user_token)\n\ntests/api/test_api_edit_user.py:89: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \napi/edit_user.py:8: in api_call\n    response = self.make_request(\"PUT\", f\"users/{username}\", json=user_dict, headers=headers)\napi/base_api.py:33: in make_request\n    response.raise_for_status()\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <Response [404]>\n\n    def raise_for_status(self):\n        \"\"\"Raises :class:`HTTPError`, if one occurred.\"\"\"\n    \n        http_error_msg = \"\"\n        if isinstance(self.reason, bytes):\n            # We attempt to decode utf-8 first because some servers\n            # choose to localize their reason strings. If the string\n            # isn't utf-8, we fall back to iso-8859-1 for all other\n            # encodings. (See PR #3538)\n            try:\n                reason = self.reason.decode(\"utf-8\")\n            except UnicodeDecodeError:\n                reason = self.reason.decode(\"iso-8859-1\")\n        else:\n            reason = self.reason\n    \n        if 400 <= self.status_code < 500:\n            http_error_msg = (\n                f\"{self.status_code} Client Error: {reason} for url: {self.url}\"\n            )\n    \n        elif 500 <= self.status_code < 600:\n            http_error_msg = (\n                f\"{self.status_code} Server Error: {reason} for url: {self.url}\"\n            )\n    \n        if http_error_msg:\n>           raise HTTPError(http_error_msg, response=self)\nE           requests.exceptions.HTTPError: 404 Client Error:  for url: http://localhost:4001/users/nonexistent_user\n\n/opt/hostedtoolcache/Python/3.12.1/x64/lib/python3.12/site-packages/requests/models.py:1021: HTTPError\n\nDuring handling of the above exception, another exception occurred:\n\nedit_user_api = <api.edit_user.EditUser object at 0x7fea9d304140>\nuser_token = 'eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ3ZHVkbGV5IiwiYXV0aCI6W3siYXV0aG9yaXR5IjoiUk9MRV9BRE1JTiJ9LHsiYXV0aG9yaXR5IjoiUk9MRV9DTElFTlQifV0sImlhdCI6MTcxODA1OTM0OSwiZXhwIjoxNzE4MDU5NjQ5fQ.QNAEY2kjsnqGwRvENJnjOMN_cD6lI3VKcbVsvtkVob0'\nnew_user_data = EditUserDto(email='newemail@example.com', firstName='NewFirstName', lastName='NewLastName', roles=['ROLE_ADMIN'])\n\n    def test_edit_user_invalid_username(edit_user_api, user_token, new_user_data):\n        try:\n            invalid_username = \"nonexistent_user\"\n            response = edit_user_api.api_call(invalid_username, new_user_data, user_token)\n            # This line should ideally never be reached if the API behaves correctly\n            assert response.status_code == 404, \"Expected status code 404 for nonexistent user\"\n        except Exception as exc:\n            assert \"404\" in str(exc), \"Expected a 404 status code for invalid username\"\n>           assert \"The user doesn't exist\" in str(exc), \"Expected error message for invalid username\"\nE           AssertionError: Expected error message for invalid username\nE           assert \"The user doesn't exist\" in '404 Client Error:  for url: http://localhost:4001/users/nonexistent_user'\nE            +  where '404 Client Error:  for url: http://localhost:4001/users/nonexistent_user' = str(HTTPError('404 Client Error:  for url: http://localhost:4001/users/nonexistent_user'))\n\ntests/api/test_api_edit_user.py:94: AssertionError","steps":[],"attachments":[{"uid":"c516d51b3806ec61","name":"log","source":"c516d51b3806ec61.txt","type":"text/plain","size":616}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":1,"hasContent":true},"afterStages":[{"name":"user_token::0","time":{"start":1718059349545,"stop":1718059349570,"duration":25},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false}],"labels":[{"name":"parentSuite","value":"tests.api"},{"name":"suite","value":"test_api_edit_user"},{"name":"host","value":"fv-az654-837"},{"name":"thread","value":"3451-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.api.test_api_edit_user"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Product defects","matchedStatuses":[],"flaky":false}],"tags":[]},"source":"961b50eabb5165e1.json","parameterValues":[]}